File: OOP_in_JAVA/Programming_Codes/Day_4/src/InheritanceAssignment/Ques5.java
// 5) create abstract class Shape. declare "draw()" function as abstract. From this class define "Triangle","Polygon" and "Circle" .
// create an array of Shape having 3 elements.
// store child objects into this array . and call
// their draw function.

package InheritanceAssignment;

//Abstract class Shape
abstract class Shape {
 // Abstract method draw
 public abstract void draw();
}

//Class Triangle extending Shape
class Triangle extends Shape {
 @Override
 public void draw() {
     System.out.println("Drawing a Triangle");
 }
}

//Class Polygon extending Shape
class Polygon extends Shape {
 @Override
 public void draw() {
     System.out.println("Drawing a Polygon");
 }
}

//Class Circle extending Shape
class Circle extends Shape {
 @Override
 public void draw() {
     System.out.println("Drawing a Circle");
 }
}

//Class with main function
public class Ques5 {
 public static void main(String[] args) {
     // Create an array of Shape with 3 elements
     Shape[] shapes = new Shape[3];

     // Store child objects into the array
     shapes[0] = new Triangle();
     shapes[1] = new Polygon();
     shapes[2] = new Circle();

     // Call the draw method for each shape
     for (Shape shape : shapes) {
         shape.draw();
     }
 }
}

File: C++_Programming/Programming_Codes/DAY-6/Assignment/Assignment/Ques5.cpp
//5) create an abstract class Shape with
//virtual void display()
//{
//	cout << "display on the canvas" << endl;
//}
//and
//pure virtual function(contract) void draw()
//functions.
//now derive 3 classes from Shape
//Circle, Triangle and Rectangle
//in the main function create array of pointer to Shape class and store all the child class objects 
// addresses inside it.Traverse the array and invoke "display()" and "draw()" functions.

//#include <iostream>
//using namespace std;
//
//class Shape {
//public:
//	virtual void display() = 0;     
//	virtual void draw() = 0;                    
//};
//
//class Circle : public Shape {
//public:
//	void display() override {
//		cout << "Displaying Circle on the canvas." << endl;
//	}
//	void draw() {
//		cout << "Drawing Circle." << endl;
//	}
//};
//
//class Triangle : public Shape {
//public:
//	void display() override {
//		cout << "Displaying Triangle on the canvas." << endl;
//	}
//	void draw() {
//		cout << "Drawing Triangle." << endl;
//	}
//};
//
//class Rectangle : public Shape {
//public:
//	void display() override {
//		cout << "Displaying Rectangle on the canvas." << endl;
//	}
//	void draw() {
//		cout << "Drawing Rectangle." << endl;
//	}
//};
//
//int main() {
//	Shape* shapes[3];
//	shapes[0] = new Circle();
//	shapes[1] = new Triangle();
//	shapes[2] = new Rectangle();
//
//	for (int i = 0; i < 3; i++) {
//		shapes[i]->display();
//		shapes[i]->draw();
//		delete shapes[i]; // Clean up memory
//	}
//	return 0;
//}